### PR TITLE
Pre-build themes using Angular CLI (sass + sass-loader) instead of node-sass Bash script

### PR DESCRIPTION
**Pull request by vazmin**
1. Use `angular-cli` instead of directly using `node-sass`
2. Use `@import '~'`

#743 #744

### CODE REVIEW

**Comment by vazmin** [comment]
@googlebot I signed it!

File: yarn.lock [hunk 1]
```diff
@@ -211,10 +211,6 @@
   version "9.2.1-sha-b0424a351"
   resolved "https://codeload.github.com/angular/material2-docs-content/tar.gz/66f276a9526a373b4c252370514f10de130561f0"
 
-"@angular/components-examples@angular/material2-docs-content#master":
```
Comment (Splaktar) [inline yarn.lock:214]: This shouldn't be removed. It looks like you rebuilt the entire lock file? If so, then you'll need to run `npm run build:content`.

**Comment by Splaktar** [comment]
Thanks for proposing these changes. They look interesting, but I need to take a look at them locally to verify how they work.

File: yarn.lock (hunk 1 above)
Comment (vazmin) [reply yarn.lock:214]: I try run 'yarn run build:content' and 'yarn upgrade @angular/components-examples', but `yarn.lock` has not changed

Comment (Splaktar) [reply yarn.lock:214]: AH OK, that makes sense. The diff didn't show that the one above it was for the same package.

**Comment by vazmin** [comment]
> The only difference I see is that the lazily loaded pre-built theme `.css` files are about `0.2 kB` larger. 

This is true in the case of `yarn run build`:

using node-sass
```css
.mat-ripple-element {
  background-color: rgba(0, 0, 0, 0.1); }
```

using angular cli
```css
.mat-ripple-element {
  background-color: rgba(0, 0, 0, 0.1);
}
```
An extra line break.

`yarn run prod-build` is the same. 

but now the site is using [unoptimized css](https://material.angular.io/assets/deeppurple-amber.css) instead of [optimized css](https://material.angular.io/deeppurple-amber.css).

Of course I also think that is negligible.

**Comment by jelbourn** [comment]
@vazmin to confirm, the prod mode CSS output it identical, and the dev mode CSS preserves whitespace?

**Comment by vazmin** [comment]
@jelbourn 

> the dev mode CSS preserves whitespace?

yes， the default options of `output-style  (nested | expanded | compact | compressed)`  are different

angular cli: 
[ outputStyle: 'expanded',](https://github.com/angular/angular-cli/blob/254994db85398d847d4ca6608c4019fc1ea7a583/packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/styles.ts#L148)

node-sass:
[ 'output-style': 'nested',](https://github.com/angular/angular-cli/blob/254994db85398d847d4ca6608c4019fc1ea7a583/packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/styles.ts#L148)

> to confirm, the prod mode CSS output it identical

build.configurations.production.optimization is true

- - -

code output before modification：
```
"assets": [
   {
      "glob": "**/*",
      "input": "src/assets",
      "output": "/assets"
   }
]
```
will output
`dist/assets/indigo-pink.css`  (unoptimized css, the site requested this)

```
{
  "inject": false,
  "input": "src/assets/indigo-pink.css",
  "bundleName": "indigo-pink"
}
```
will output
`dist/indigo-pink.css` (optimized css)

- - -

code output before modification：
```
{
  "lazy": true,
  "inject": false,
  "input": "src/styles/custom-themes/indigo-pink.scss",
  "bundleName": "assets/indigo-pink"
}
```
will output
`dist/assets/indigo-pink.css`  (optimized css)

The `intellij idea` I use has the function` Compare with` which provides a comparison file, the content of the optimized css is the same,

File: angular.json [hunk 2]
```diff
@@ -21,6 +21,7 @@
             "index": "src/index.html",
             "main": "src/main.ts",
             "tsConfig": "tsconfig.app.json",
+            "extractCss": true,
```
Comment (devversion) [inline angular.json:24]: We could remove `extractCss` from the production configuration, right?

Comment (vazmin) [reply angular.json:24]: Yes, `build.options.extractCss ` is already true, ` build.configurations.production.extractCss` can be removed.

**Pull request merged**